packages/@okta/vuepress-site/docs/guides/mobile-idx-sdk-overview/main/index.md

```diff
@@ -0,0 +1,165 @@
+
+![A diagram showing the SDK objects and their relationships.](/img/mobile-sdk/mobile-idx-objects.png "A diagram that shows the SDK objects for the sign in flow and the relationships between them.")
+
+</div>
+
+## Common parts of the flow
+
+Some parts of the sign-in flow are the same:
+
+- Add the SDK to your app.
+- Configure the SDK.
+- Initialize the client and start the flow.
+- Process the response.
+- Request a token.
+- Sign-out the user.
+
+### Add the SDK
+
+<StackSnippet snippet="adddependency" />
+
+### Create and manage configurations
+
+A _Configuration_ contains the settings used by the SDK to connect to an Okta Application Integration:
+
+| Value         | Description |
+| :------------ | :---------- |
+| Issuer        | The Oauth2 URL for the Okta Org, such as `https://oie-123456.oktapreview.com/oauth2/default`. |
+| Client ID     | The ID of the Okta Application Integration from the Okta Admin console.  |
+| Client secret | An optional shared secret for accessing the Application Integration. |
+| Redirect URI  | A callback URI for launching the mobile app, such as `com.example.oie-123456:/callback`. |
+| Scopes        | A space separated list of the requested access scopes for the app. |
+
+The configuration information in a shipping app is usually static. You can initialize the
+configuration values directly in the code or by reading them in from a file. During
+development you may want to provide a way to edit configuration values.
+
+<StackSnippet snippet="loadingaconfiguration" />
+
+### Start sign-in
+
+Start the sign-in flow by first creating an SDK client using a configuration, and then requesting the first step.
+
+<StackSnippet snippet="initializingsdksession" />
+
+### Process the response
+
+The steps for processing a response are:
+1. Check for a successful sign-in.
+1. Check for messages, such as an invalid password.
+1. Check for remediations.
+1. Process the remediations.
+1. Process the authenticators.
+
+After the user enters the required information, update the remediation and request the next step.
+
+<StackSnippet snippet="processresponse" />
+
+### Complete sign in
+
+Check the response for a successful sign-in at each step of the flow. When the user is
+signed-in, exchange the session token for an access token, and then exit the flow.
+
+<StackSnippet snippet="gettingatoken" />
+
+### Sign out a user
+
+Sign out a user by revoking the access token granted by signing in, and by revoking a
+refresh token if one exists.
+
+<StackSnippet snippet="signingout" />
+
+## Sign-in flow UI
+
+There are two general approaches for adding the sign-in flow to your app: add a fixed
+number of authentication methods possibly in a fixed order, or generate the user interface
+dynamically based on the response at each step.
+
+For a fixed number of methods, you create appropriate views and populate them from the
+response. Using fixed methods does introduce risk as authentication methods may be added
```

Comment:
We should emphasize the risk introduced via this approach! (maybe make it red or bold or something)

Reply:
First it's important to understand that a developer reading this document either hasn't implemented Okta in their app, or has, but it isn't going the way they want. They don't have any of the background in the SDK that we have or any understanding of the true complexity involved.

I had originally considered a Note or Warning, but doing that looses the flow of the "story", of the journey that we're leading the developer through. If you go through the article you see that I build to this "climax":
You can add sign-in in two ways: let Okta handle it or do it yourself.
(choose do it yourself)
(start slow...)
Policies control how users' sign-in
Admins (not the app developer) control how many and which ones
There's lots so there's lots of combinations
Build your app to step through the different access methods
Here's how those steps are represented (Sign-in objects)
Here's some common code (Common parts of the flow and it's heading)
(they finish the common code and in the back of their mind is "but how do I parse remediations and authenticators" into a UI, hence the final section...)
The UI section starts by bringing static and dynamic implementations top of mind (some developers will never think of dynamic until now)
It then builds the "do it dynamically" case by showing more and more risk with the static way: the 2nd paragraph introduces "some risk", the bullet points start to make it clear how hard it is to update apps, the 3rd paragraph deals with "but it's an enterprise app", and the 4th drives the point home.

If I want to emphasize a point in our docs my only choice is a Note or perhaps Warning box (per the style guide.) Both of those make it really hard to lead the developer through the thought process of "oh, I should do dynamic" because they interrupt that flow.

And yes, I really do think through this kind of stuff ;-)

Based on that, are you OK with the way it is?